### PR TITLE
Add easy-to-use options to storySort

### PR DESCRIPTION
**Pull request by JohnAlbin**
Issue: #9185

## What I did

Adds the `storySort` object parameter described in #9185.

## How to test

- Tests are included
- An existing example is updated
- Documentation is updated

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -256,7 +256,7 @@ For example, here's how to sort by story ID using `storySort`:
 addParameters({
   options: {
     storySort: (a, b) =>
-      a[1].kind === b[1].kind ? 0 : a[1].id.localeCompare(b[1].id, { numeric: true }),
+      a[1].kind === b[1].kind ? 0 : a[1].id.localeCompare(b[1].id, undefined, { numeric: true }),
```
Comment (JohnAlbin) [inline MIGRATION.md]: FYI, in [localeCompare()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare), the `options` array is the 3rd parameter, not the 2nd one.

Comment (shilman) [reply MIGRATION.md]: 😱

**Comment by shilman** [comment]
This looks pretty awesome @JohnAlbin! 🙌 

We're out of time on 5.3, but it's now in the queue for 6.0 which will start in early Jan. Thanks for the contribution ... and being patient!

**Comment by JohnAlbin** [comment]
> We're out of time on 5.3, but it's now in the queue for 6.0 which will start in early Jan.

Yep. I saw the announcement in Discord this week. No worries.

File: MIGRATION.md (hunk 1 above)
Comment (ndelangen) [reply MIGRATION.md]: 🙀

**Review by ndelangen — APPROVED** [review]

File: docs/src/pages/configurations/options-parameter/index.md [hunk 2]
```diff
@@ -86,9 +88,65 @@ addParameters({
...
+The `storySort` can also accept a configuration object.
+
+```js
+import { addParameters, configure } from '@storybook/react';
+
+addParameters({
+  options: {
+    storySort: {
+      method: 'alphabetical', // Optional, defaults to 'configure'.
+      sort: ['Intro', 'Components'], // Optional, defaults to [].
```
Comment (chriscarpenter12) [inline docs/src/pages/configurations/options-parameter/index.md]: I see the PR merged in `next-6.0.0`, but not seeing the changes reflected there. I wanted to point out the `sort` key here is wrong. The `storySort` method in this PR uses `order` to override the order.

Comment (shilman) [reply docs/src/pages/configurations/options-parameter/index.md]: Cc @ndelangen

Comment (hasparus) [reply docs/src/pages/configurations/options-parameter/index.md]: Yeah, this doc seems to be outdated. There's `order` in tests.

Comment (ndelangen) [reply docs/src/pages/configurations/options-parameter/index.md]: @hasparus would you be able to make a PR fixing the docs for this?

Comment (hasparus) [reply docs/src/pages/configurations/options-parameter/index.md]: yup, no biggie

**Pull request merged**